wrap.go

```diff
@@ -0,0 +1,121 @@
+package batcher
+
+import (
+	"context"
+	"fmt"
+)
+
+type wrapper[K comparable, V any] struct {
```

Comment:
Maybe it's better to call it a middleware and move it to a separate package?

Reply:
it looks like middleware, but if I call it like middleware, then I need to create the ability to add custom middleware. I definitely don't want it